spec-traits/src/main/kotlin/traits/DataKindTrait.kt

```diff
@@ -0,0 +1,80 @@
+package traits
+
+import software.amazon.smithy.model.SourceException
+import software.amazon.smithy.model.node.Node
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.traits.AbstractTrait
+import software.amazon.smithy.model.traits.AbstractTraitBuilder
+import software.amazon.smithy.model.traits.TraitService
+import software.amazon.smithy.utils.SmithyBuilder
+import software.amazon.smithy.utils.ToSmithyBuilder
+import java.util.stream.Collectors
+
+class DataKindTrait(builder: Builder) : AbstractTrait(ID, builder.sourceLocation), ToSmithyBuilder<DataKindTrait> {
```

Comment:
Here again how is it better to translate from Java to Kotlin?

Reply:
it's much much shorter